security-setup

```diff
@@ -449,17 +465,18 @@ class Mesos(Component):  # Mesos should always come after any frameworks
                 print('removed mesos_authenticate_followers. This value is now automatically generated - see the docs.')
 
             # remove follower credential from credential list
-            try:
-                credential_idx = config['mesos_credentials'].index({
-                    'principal': config['mesos_follower_principal'],
-                    'secret': config['mesos_follower_secret']
-                })
-            except ValueError:
-                credential_idx = None
-
-            if credential_idx is not None:
-                del config['mesos_credentials'][credential_idx]
-                print('removed follower credentials from credential list - it is now a separate setting.')
+            if not self.args.no_authenticate_followers and not self.args.no_authenticate_frameworks:
```

Comment:
this setting is actually part of a deprecation and should be removed regardless of those flags.